martha/martha_v3.js

```diff
@@ -1,36 +1,156 @@
-const { dataObjectUriToHttps, parseRequest, convertToMarthaV3Response, FailureResponse } = require('../common/helpers');
-const { maybeTalkToBond, determineBondProvider, BondProviders } = require('../common/bond');
+const { parseRequest, convertToMarthaV3Response, FailureResponse } = require('../common/helpers');
+const config = require('../config.json');
 const apiAdapter = require('../common/api_adapter');
+const url = require('url');
 
 const BAD_REQUEST_ERROR_CODE = 400;
 const SERVER_ERROR_CODE = 500;
 
+class DrsType {
+    constructor(drsUrl, sendAuth, bondUrl, responseParser) {
+        this.drsUrl = drsUrl;
+        this.sendAuth = sendAuth;
+        this.bondUrl = bondUrl;
+        this.responseParser = responseParser;
+    }
+}
 
-function validateRequest(dataObjectUri, auth) {
-    if (!dataObjectUri) {
-        throw new Error('URL of a DRS object is missing.');
-    } else if (!auth) {
-        throw new Error('Authorization header is missing.');
+/** *************************************************************************************************
+ * URI parsers
+ */
+
+function dosFullUrlGenerator (parsedUrl) {
+    return url.format({
+        protocol: 'https',
+        hostname: parsedUrl.hostname,
+        port: parsedUrl.port,
+        pathname: `/ga4gh/dos/v1/dataobjects${(parsedUrl.pathname) ? parsedUrl.pathname : `/${parsedUrl.hostname}`}`,
+        search: parsedUrl.search
+    });
+}
+
+function dosCompactUrlGenerator (parsedUrl) {
```

Comment:
(very knit-picky detail, but as long as I'm writing other comments) There is a space between the function name and the `(`. This doesn't seem to be the style in the rest of the file but is the style for all newly added functions.  I might suggest consistency over personal preference (going a step further, I'd suggest installing `Prettier` on this codebase so that it auto formats and people never get into style arguments ever).

Reply:
This seems like it should be a new Jira issue (e.g. installing Prettier).

Reply:
I completely agree! Again, I only added this comment since I was already commenting to say 2 things: 1. I think we should have consistent style, so I would recommend removing those blank spaces next time you touch the code.  And 2. maybe install Prettier in the future (*not* now :) if you think it'd save time (I do).  All of this is extremely low level of importance!